atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java

```diff
@@ -82,29 +85,33 @@ public void testRender() throws IOException {
     @Test
     public void testAddDelete() throws Exception {
         final byte[] data = PtBytes.toBytes("streamed");
-        final long streamId = txManager.runTaskWithRetry(new TransactionTask<Long, Exception>() {
-            @Override
-            public Long execute(Transaction t) throws Exception {
-                PersistentStreamStore store = StreamTestStreamStore.of(txManager, StreamTestTableFactory.of());
-                byte[] data = PtBytes.toBytes("streamed");
-                Sha256Hash hash = Sha256Hash.computeHash(data);
-                byte[] reference = "ref".getBytes();
-                long streamId = store.getByHashOrStoreStreamAndMarkAsUsed(t, hash, new ByteArrayInputStream(data), reference);
-                try {
-                    store.loadStream(t, 1L).read(data, 0, data.length);
-                } catch (NoSuchElementException e) {
-                    // expected
-                }
-                return streamId;
-            }
+        final long streamId = txManager.runTaskWithRetry((TransactionTask<Long, Exception>) t -> {
+            PersistentStreamStore store = StreamTestStreamStore.of(txManager, StreamTestTableFactory.of());
+            Sha256Hash hash = Sha256Hash.computeHash(data);
+            byte[] reference = "ref".getBytes();
+
+            return store.getByHashOrStoreStreamAndMarkAsUsed(t, hash, new ByteArrayInputStream(data), reference);
         });
-        txManager.runTaskWithRetry(new TransactionTask<Void, Exception>() {
-            @Override
-            public Void execute(Transaction t) throws Exception {
-                PersistentStreamStore store = StreamTestStreamStore.of(txManager, StreamTestTableFactory.of());
-                Assert.assertEquals(data.length, store.loadStream(t, streamId).read(data, 0, data.length));
-                return null;
-            }
+        txManager.runTaskWithRetry((TransactionTask<Void, Exception>) t -> {
+            PersistentStreamStore store = StreamTestStreamStore.of(txManager, StreamTestTableFactory.of());
+            Optional<InputStream> inputStream = store.loadSingleStream(t, streamId);
+            assertTrue(inputStream.isPresent());
+            assertEquals(data.length, inputStream.get().read(data, 0, data.length));
```

Comment:
I realize this test was going this previously, but this should really be validating the contents of the data read and also do a `ByteStreams.readFully(inputStream.get(), data)`, to ensure the full stream is read (though I think for this test read doesn't return less than requested length, but might in a compressed or encrypted scenario.